### PR TITLE
Create slurm cluster

### PR DESCRIPTION
**Pull request by YidingZhou**

### CODE REVIEW

File: slurm/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "West US",
+            "metadata": {
+                "description": "This is the location where the availability set will be deployed"
+            }
+        },
+        "MasterVMName": {
```
Comment (singhkay) [inline slurm/azuredeploy.json]: camelCase for these parameter names

File: slurm/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+            "type": "string",
+            "defaultValue": "master",
+            "metadata": {
+                "description": "The machine name of the master node - not be confused with dnsName which is public visible name."
+            }
+        },
+        "WorkerVMName": {
```
Comment (singhkay) [inline slurm/azuredeploy.json]: camelCase for these parameter names

File: slurm/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+                        "[concat(variables('templateBaseUrl'), 'azuredeploy.sh')]"
+                    ],
+                    "commandToExecute": "[variables('installationCLI')]"
+                }
+            }
+        },
+        {
+            "apiVersion": "2015-05-01-preview",
+            "type": "Microsoft.Network/networkInterfaces",
+            "name": "[concat(variables('nicName'), 'worker', copyindex())]",
```
Comment (singhkay) [inline slurm/azuredeploy.json]: Did you mean to use the nicname here?

File: slurm/metadata.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{ 
+  "itemDisplayName": "Spin up a SLURM cluster", 
+  "description": "Template spins up a SLURM cluster.", 
+  "githubUsername": "YidingZhou", 
+  "summary": "SLURM cluster",
+  "dateUpdated": "2015-05-18"
```
Comment (singhkay) [inline slurm/metadata.json]: Update this to a more recent date :)

**Comment by singhkay** [comment]
Adding @johngossman for review as well

**Comment by YidingZhou** [comment]
Updated date and camelCase.

File: slurm/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "West US",
+            "metadata": {
+                "description": "This is the location where the availability set will be deployed"
+            }
+        },
+        "masterVMName": {
```
Comment (johngossman) [inline slurm/azuredeploy.json]: Don't make this a parameter. Just hardcode "master"

File: slurm/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+            "type": "string",
+            "defaultValue": "master",
+            "metadata": {
+                "description": "The machine name of the master node - not be confused with dnsName which is public visible name."
+            }
+        },
+        "workerVMName": {
```
Comment (johngossman) [inline slurm/azuredeploy.json]: Don't make this a parameter

File: slurm/azuredeploy-parameters.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (johngossman) [inline slurm/azuredeploy-parameters.json]: This file should match the parameters in the template

File: slurm/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+                "description": "Unique public dns name where the master node will be exposed"
+            }
+        },
+        "newStorageAccountName": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+            }
+        },
+        "adminPassword": {
```
Comment (johngossman) [inline slurm/azuredeploy.json]: Put this parameter next to adminUser

File: slurm/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+            "type": "securestring",
+            "metadata": {
+                "description": "Admin password"
+            }
+        },
+        "vmSize": {
```
Comment (johngossman) [inline slurm/azuredeploy.json]: Provide allowedValues for this. Use D-sizes instead of A-sizes

**Comment by johngossman** [comment]
Could you please squash your commits?

**Comment by YidingZhou** [comment]
I've squashed the request. How do eliminate the acomghbot comment?

**Comment by YidingZhou** [comment]
Squashed all commits and updated according to comments. #1 VM size, #2 Move VM name from parameters to variables, #3 Co-located username and password parameters, and #4 make parameter json match the deployment template.

File: slurm/README.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# Deploy a slurm cluster
+
+Built by: [YidingZhou](https://github.com/YidingZhou)
+
+##Deploy
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FYidingZhou%2Fazure-quickstart-templates%2Fmaster%2Fslurm%2Fazuredeploy.json" target="_blank">
```
Comment (singhkay) [inline slurm/README.md]: This needs to be updated to point to this repo

File: slurm/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+                    "base": "10.0.0.",
+                    "start": 5
+                },
+                "master": "10.0.0.254"
+            }
+        },
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('networkSettings').virtualNetworkName)]",
+        "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('networkSettings').subnet.dse.name)]",
+        "sshKeyPath": "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+        "templateBaseUrl": "https://raw.githubusercontent.com/YidingZhou/azure-quickstart-templates/master/slurm/",
```
Comment (singhkay) [inline slurm/azuredeploy.json]: This needs to be updated for this repo as well

File: slurm/azuredeploy.sh [hunk 12]
```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+# This script can be found on https://github.com/YidingZhou/azure-quickstart-templates/blob/master/slurm/azuredeploy.sh
```
Comment (singhkay) [inline slurm/azuredeploy.sh]: This url needs update as well

**Pull request merged**